aten/src/ATen/core/ivalue_inl.h

```diff
@@ -429,6 +437,21 @@ std::vector<Elem> generic_to(
   }
   return result;
 }
+template <typename T>
+T generic_to(
+    IValue ivalue,
+    _fake_type<T>) {
+    using ElemType = typename T::element_type;
+    auto obj = ivalue.toObject();
+    auto capsule = obj->getAttr("capsule");
+    auto capsulePtr = capsule.toCapsule().release();
+    if (capsulePtr == nullptr) {
+      capsulePtr = (ElemType*)malloc(sizeof(ElemType));
+    }
+    auto initializedCapsule = IValue(c10::intrusive_ptr<c10::intrusive_ptr_target>::reclaim(static_cast<intrusive_ptr_target*>(capsulePtr)));
```

Comment:
@zdevito I'm not sure I'm doing this correctly here, since it's possible that `capsulePtr` is a malloc'ed pointer.

Reply:
Resolved.